src/main/java/com/github/manolo8/darkbot/utils/login/LoginUtils.java

```diff
@@ -194,6 +208,8 @@ private static String replaceParameters(String params) {
     }
 
     private static String getLoginUrl(String in) {
+        System.out.println("captcha: " + in.contains("recaptcha_image"));
+        captcha = in.contains("recaptcha_image");
```

Comment:
What is this doing here? Why does a function to **get the login url from the string page** cause a side-effect in the global "captcha" variable?
Remove this from here, and perform this check where it belongs, not by abusing something else, you have the "frontPage" variable pretty clearly available